.gitignore

```diff
@@ -1 +1,28 @@
 bin/
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
+*.o
+*.a
+*.so
```

Comment:
where did all of this stuff come from? was it autogenerated? Most (all?) of this is irrelevant